dbms/src/Parsers/ParserInsertQuery.cpp

```diff
@@ -97,6 +117,13 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
         pos = before_select;
         ParserSelectWithUnionQuery select_p;
         select_p.parse(pos, select, expected);
+
+        /// Check if we have INSERT SELECT FROM input().
+        tryFindInputFunction(select, input_function);
+        /// FORMAT section is required if we have input() in SELECT part
+        if (input_function)
```

Comment:
It's not a syntax error if we have a query with input but without FORMAT.
Better to allow it to parse but check later (and provide reasonable error message).